### ISSUE TITLE
snnTorch population-coded accuracy can mis-score close float64 predictions

### ISSUE THREAD

**Issue by GraceDyer** [jeshraghian/snntorch#429]
## Summary

I found a reproducible issue in `snntorch.functional.acc.accuracy_rate` when using `population_code=True`. For close `float64` population-coded class scores, the public accuracy helper can report the wrong class even though the target class has the larger score.

## Minimal Reproduction

```python
import torch
from snntorch.functional import acc

spk_out = torch.tensor([[[1.00000002, 1.00000005]]], dtype=torch.float64)
targets = torch.tensor([1])

accuracy = acc.accuracy_rate(
    spk_out,
    targets,
    population_code=True,
    num_classes=2,
)
pop_code = acc._population_code(spk_out, num_classes=2, num_outputs=2)

print("accuracy:", accuracy)
print("population_code:", pop_code)

assert accuracy == 1.0
```

## Expected Behavior

The population-coded metric should preserve the class ordering implied by the input scores, or explicitly document/cast precision in a way that does not silently change the predicted class.

## Actual Behavior

The assertion fails. In my reproduction, the helper returned:

```text
accuracy: 0.0
population_code: tensor([[1., 1.]])
```

The close `float64` score difference is lost, so the metric reports the wrong result.

**Comment by ixfd64** [comment]
Should be resolved in #434 (thanks to @rlogger). Feel free to re-open this if there are still issues.
